### PR TITLE
ember-cli-coffee-snippets

### PR DESCRIPTION
**Pull request by imanghafoori1**
These snippets are optimized for and written with ember-cli and coffeeScript in mind.

### CODE REVIEW

File: snippets/coffee-adapters.cson [hunk 1]
```diff
@@ -0,0 +1,28 @@
+'.source.coffee':
+
+  'Ember fixtureAdapter':
+    'prefix': 'ds.fixA'
```
Comment (seifsallam) [inline snippets/coffee-adapters.cson]: Can you use more clear prefix like `fixture-adapter`, so it's similar to the rest of the rest of the project. Also please apply this concept to the rest of the prefixes. `rest-adapter`

Can you also move adapters to `modules.cson`

File: snippets/coffee-adapters.cson [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    'body':
+      """
+        export = DS.FixtureAdapter.extend()
+      """
+
```
Comment (seifsallam) [inline snippets/coffee-adapters.cson]: Can you remove extra spacing. Only 1 line between is enough.

File: snippets/coffee-adapters.cson [hunk 3]
```diff
@@ -0,0 +1,28 @@
+'.source.coffee':
+
+  'Ember fixtureAdapter':
+    'prefix': 'ds.fixA'
+    'body':
+      """
+        export = DS.FixtureAdapter.extend()
```
Comment (seifsallam) [inline snippets/coffee-adapters.cson]: Can you add body here, like the rest of your methods

File: snippets/coffee-ember-object [hunk 4]
```diff
@@ -0,0 +1,243 @@
+
+'.source.coffee':
+
+  'Ember Route':
+    'prefix': 'em.rou'
```
Comment (seifsallam) [inline snippets/coffee-ember-object]: Same rule for prefixes too. Please follow my prefixes naming

File: snippets/coffee-ember-object [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+    'body':
+      """
+        _export = Ember.Route.extend(
+          model: ->
+            ${1:#@store.find("name_of_a_model")}
```
Comment (seifsallam) [inline snippets/coffee-ember-object]: `name_of_a_model` -> `model_name`

File: snippets/coffee-ember-object [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+
+      """
+
+  'Ember Array Controller':
+    'prefix': 'em.arr'
+    'body':
+      """
+        _export = Ember.ArrayController.extend(
+          ${1:itemController: 'name_of_an_objectController'}
+          ${2:sortProperties: ['','']}
```
Comment (seifsallam) [inline snippets/coffee-ember-object]: I feel like this is too complex. Why sorting and why it requires two actions?

File: snippets/coffee-ember-object [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+          ${4:actions:
+            ${5:name}: ->
+              ${6:#logic...}
+            ${7:name2}: ->
+              ${8:#logic2...}
+            }
+        )
+      """
+
+
```
Comment (seifsallam) [inline snippets/coffee-ember-object]: Please remove extra lines. Only 1 line should do

File: snippets/coffee-ember-object [hunk 8]
```diff
@@ -0,0 +1,243 @@
...
+      """
+        _export = Ember.View.extend(
+          tagName: ${1:"div"}
+          ${2:classNames: ['','']}
+          ${3:classNameBindings: ['','']}
+          ${4:eventManager: Ember.Object.create(
+            mouseEnter: (event, view) ->
+              #logic...
+              return
+            click:      (event, view) ->
```
Comment (seifsallam) [inline snippets/coffee-ember-object]: Why mouseEnter and click?

File: snippets/coffee-ember-object [hunk 9]
```diff
@@ -0,0 +1,243 @@
+
+'.source.coffee':
```
Comment (seifsallam) [inline snippets/coffee-ember-object]: Please move merge this file's content with existing snippets based on it's type (function, module, property)

File: snippets/coffee-ember-object [hunk 10]
```diff
@@ -0,0 +1,243 @@
...
+  'Ember Model':
+    'prefix': 'ds.mod'
+    'body':
+      """
+        _export = DS.Model.extend(
+          ${1:title}: DS.attr(${2:"string"})
+          ${3:title: DS.attr("boolean")}
+          ${4:title: DS.attr("number")}
+          ${5:title: DS.belongsTo('someModel',async: true)}
+          ${6:title: DS.hasMany('someModel'  ,async: true)}
```
Comment (seifsallam) [inline snippets/coffee-ember-object]: I don't see need for adding all this as part of model. I believe it's better to have it's separate autocomplete.

File: snippets/coffee-functions.cson [hunk 11]
```diff
@@ -0,0 +1,67 @@
+# Ember Functions snippets
+#
+# Examples
+#
+#  `ember<tab>`
+#
+
+'.source.coffee':
```
Comment (seifsallam) [inline snippets/coffee-functions.cson]: Please move merge this file's content with existing snippets file

File: snippets/es6-syntax.cson [hunk 12]
```diff
@@ -0,0 +1,17 @@
+'.source.coffee':
+
+  'ES6 import syntax':
+    'prefix': 'es.imp'
```
Comment (seifsallam) [inline snippets/es6-syntax.cson]: I think there should be separate prefixes `es.imp` -> `import-ember` and `es.imp` -> `import-data`

**Comment by mattmcmanus** [comment]
Hey @imanghafoori1. These PR's have been sitting for a while and I'm going close them. I've decided to deprecate coffeescript support as it's not the community standard at this point. I'll keep around the existing snippets but will not be updating them.

**Pull request closed without merging**